connectors/http/http-base/src/main/java/io/camunda/connector/http/base/request/apache/bulider/parts/ApacheRequestBodyBuilder.java

```diff
@@ -0,0 +1,116 @@
+
+  @Override
+  public void build(ClassicRequestBuilder builder, HttpCommonRequest request)
+      throws JsonProcessingException {
+    if (request.getMethod().supportsBody && request.hasBody()) {
+      unescapeBody(request);
+
+      switch (request.getBody()) {
+        case Map<?, ?> body -> {
+          if (isFormUrlEncoded(request)) {
+            setUrlEncodedFormEntity(body, builder);
+          } else {
+            setStringEntity(builder, request);
+          }
+        }
+        case List<?> ignored -> setStringEntity(builder, request);
+        case String ignored -> setStringEntity(builder, request);
+        default ->
+            throw new IllegalStateException(
+                "Unexpected value type for the request body. Excepting Map, List, or String, got "
+                    + request.getBody().getClass().getSimpleName());
+      }
+    }
+  }
+
+  private Optional<String> tryGetContentType(HttpCommonRequest request) {
+    return Optional.ofNullable(request.getHeaders()).map(headers -> headers.get(CONTENT_TYPE));
+  }
+
+  private boolean isFormUrlEncoded(HttpCommonRequest request) {
+    return tryGetContentType(request)
+        .map(
+            s ->
+                ContentType.parse(s)
+                    .getMimeType()
+                    .equalsIgnoreCase(APPLICATION_FORM_URLENCODED.getMimeType()))
+        .orElse(false);
+  }
+
+  private void setStringEntity(ClassicRequestBuilder requestBuilder, HttpCommonRequest request)
+      throws JsonProcessingException {
+    Object body = request.getBody();
+    Optional<String> contentType = tryGetContentType(request);
+    requestBuilder.setEntity(
+        body instanceof String s
+            ? new StringEntity(
+                s,
+                contentType
+                    .map(ContentType::parse)
+                    .orElse(ContentType.TEXT_PLAIN.withCharset(StandardCharsets.UTF_8)))
+            : new StringEntity(
+                ConnectorsObjectMapperSupplier.DEFAULT_MAPPER.writeValueAsString(body),
+                contentType
+                    .map(ContentType::parse)
+                    .orElse(ContentType.APPLICATION_JSON.withCharset(StandardCharsets.UTF_8))));
+  }
+
+  private void setUrlEncodedFormEntity(Map<?, ?> body, ClassicRequestBuilder requestBuilder) {
+    requestBuilder.setEntity(
+        HttpEntities.createUrlEncoded(
+            body.entrySet().stream()
+                .map(
+                    e ->
+                        new BasicNameValuePair(
+                            String.valueOf(e.getKey()), String.valueOf(e.getValue())))
+                .collect(Collectors.toList()),
+            StandardCharsets.UTF_8));
+  }
+
+  private void unescapeBody(HttpCommonRequest request) {
+    if (request.getBody() instanceof String) {
+      String unescapedBody = StringEscapeUtils.unescapeJson((String) request.getBody());
+      request.setBody(unescapedBody);
+    }
```

Comment:
Might also consider unescaping header values. Although not directly related, [this issue](https://github.com/camunda/connectors/issues/2659) shows that sometimes there might be extra escape characters there too